### PR TITLE
Merge Fixes

### PR DESCRIPTION
**Pull request by TiagoBrenck**
## Purpose
After merging the branch `PrepareForMSIdentityNugetPackage`, I ran into the sample to test them and I found a few issues, fixed in this PR.

The fixes are:
- I have removed DefaultMapInboundClaims code since we will discuss about it separately and address it in a separate PR with all the necessary changes (it was buggy)
- Step 4 was missing the config to `AddHttpContextAccessor`
- The `WebApiServiceCollectionExtensions.cs` had the code `services.Configure<AzureADOptions>(options => configuration.Bind(configSectionName, options))`. It was conflicting with `JwtBearerOptions` and since we didnt have this config before the merge, I have removed it and the bug was fixed. 
- We had an `appsetting.json` commited with value, so I reset it

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Simply create the step 4 sample, and run it.

## What to Check
- You will be able to run the sample and consume the API
- The TodoList owner is the same name as the logged user shown on the top menu bar.

### CODE REVIEW

File: 4-WebApp-your-API/TodoListService/Startup.cs [hunk 1]
```diff
@@ -51,13 +51,11 @@ public void ConfigureServices(IServiceCollection services)
             // By default, the claims mapping will map claim names in the old format to accommodate older SAML applications.
             // 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' instead of 'roles'
             // This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
-            JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+            //JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+
+            services.AddHttpContextAccessor();
```
Comment (jmprieur) [inline 4-WebApp-your-API/TodoListService/Startup.cs]: I think that this should be in AddProtectedWebApi, not in Startup.cs ?

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -41,13 +41,9 @@ public static class WebApiServiceCollectionExtensions
             string configSectionName = "AzureAD",
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
         {
-            services.Configure<AzureADOptions>(options => configuration.Bind(configSectionName, options));
```
Comment (jmprieur) [inline Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: Why did you remove this one, @TiagoBrenck  ?

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -41,13 +41,9 @@ public static class WebApiServiceCollectionExtensions
...
             services.AddAuthentication(AzureADDefaults.JwtBearerAuthenticationScheme)
                     .AddAzureADBearer(options => configuration.Bind(configSectionName, options));
 
-            // Add session if you are planning to use session based token cache , .AddSessionTokenCaches()
```
Comment (jmprieur) [inline Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: I agree.
@kalyankrishna1 : does it make sense to have a session in Web apis?

File: 4-WebApp-your-API/TodoListService/Startup.cs [hunk 4]
```diff
@@ -51,13 +51,11 @@ public void ConfigureServices(IServiceCollection services)
...
 
             services.AddProtectedWebApi(Configuration);
-            services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
```
Comment (jmprieur) [inline 4-WebApp-your-API/TodoListService/Startup.cs]: I thought that we want to replace this one with options.TokenValidationParameters.NameClaimType = "prefered_username" ?

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs (hunk 2 above)
Comment (TiagoBrenck) [reply Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: If you add this line back, you will notice that the `JwtBearerOptions` doesn't get set and the sample doesn't work. I compared with the version before merge and I noticed that we didn't have this line, so after removing it and seeing that the sample worked, I assumed that this line was the problem.

File: 4-WebApp-your-API/TodoListService/Startup.cs (hunk 4 above)
Comment (TiagoBrenck) [reply 4-WebApp-your-API/TodoListService/Startup.cs]: For that to work, we will need to add a manual step on Configure.ps1 to change the manifest property `"accessTokenAcceptedVersion": 2` and also have this step on the README, when the developer creates the application via portal. 
Is that fine? If it is, I would address this in another PR so all of our samples issue v2 tokens.
@kalyankrishna1, @jmprieur what do you think?

Comment (jmprieur) [reply 4-WebApp-your-API/TodoListService/Startup.cs]: Yes, I think that the Web API in our samples should accept v2.0 tokens.

File: 4-WebApp-your-API/TodoListService/Startup.cs (hunk 1 above)
Comment (jmprieur) [reply 4-WebApp-your-API/TodoListService/Startup.cs]: @TiagoBrenck : on this one, we don't want apps to have to care about implementation details. .AddProtectedWebApi() needs to AddHttpContextAccessor if it uses it (I'm not sure what for, though?)

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs (hunk 2 above)
Comment (jmprieur) [reply Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: That makes sense. There is already the .AddAzureADBearer.

File: 4-WebApp-your-API/TodoListService/Startup.cs (hunk 1 above)
Comment (TiagoBrenck) [reply 4-WebApp-your-API/TodoListService/Startup.cs]: moved to the library.

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs (hunk 3 above)
Comment (TiagoBrenck) [reply Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: Kalyan suggested me to remove this line (and it is even commented).

File: 4-WebApp-your-API/TodoListService/Startup.cs (hunk 4 above)
Comment (TiagoBrenck) [reply 4-WebApp-your-API/TodoListService/Startup.cs]: As discussed on our weekly meeting, we will wait for powershell to accepts changing the accepted token version to address this name claims topics. So for now we will keep as is (the step 4 sample wont mention about token versions so it wont have preferred_username) and the `User.Name` will be getting from `name` claim.

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs (hunk 3 above)
Comment (JasonKleban) [reply Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: Because I'm trying to minimize Azure hosting costs, I wish to have all necessary state stored in cookies rather than back the session/token store with a database or distributed cache.  Does this change basically require such a persistence?  If all I need to do is know the user's verified identity, is cookie storage impractical?

Following the 1-WebApp-OIDC sample verbatim, I'm able to log in and navigate for a while, but after some time - a few minutes or a few hours, not sure yet, but definitely not days.

Is it upon AppService idle?  Manual AppService restart (probably, but the timing is not perfectly aligned)?  Session Timeout?  Are session keys being reset and no longer valid when Azure re-balances where the app service is hosted?  I'm having trouble diagnosing it.  Trying `services.Configure<SessionOptions>(options => { options.IdleTimeout = TimeSpan.FromMinutes(1); });` seems to have no effect - I'm not logged out after 1 minute of zero activity.

At such times of being unexpectedly logged out, `User.Identity.IsAuthenticated` is false.  My application attempts an interactive login, but AzureAD redirects back immediately and logs me in.  This would be less noticeable/disruptive if I weren't in an iframe requiring a popup window for the login sequence.  (I'm pretty sure the popup behavior has nothing to do with being forgotten early.)

I'm sure I'm missing some important concepts, but there seems to be a lot of fact collisions between non-core and core and between versions and among alternatives to this AzureSamples/Microsoft.Identity.Web approach.  It's hard to know what documentation/SO posts are relevant for this scenario.

Comment (jmprieur) [reply Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: @JasonKleban : if you are building a Web API (not a web app), the sample to have a look to is https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2.
